packages/reporter-vue/src/header/ReporterHeader.spec.js

```diff
@@ -1,48 +1,53 @@
 import ReporterHeader from './ReporterHeader.vue'
-import { useStatsStore } from '../store/reporter-store'
-// import { mount } from '@cypress/vue'
 import { h } from 'vue'
-import VTooltip from 'v-tooltip'
-
-const style = {
-  resize: 'both',
-  overflow: 'auto',
-  display: 'block',
-  color: 'black',
-  background: 'rgba(100,100,100, 0.1)'
-}
-
-it.only('renders', () => {
-  const timeout = 1512
-  const wrapper = {
-    setup() {
-      // const store = useStatsStore()
-      // // const reporter = useReporterStore()
+import { useStatsStore } from '../store/reporter-store'
 
-      // cy.spy(reporter, 'rerun').as('rerun clicked')
-      // cy.spy(reporter, 'toggleAutoScrolling').as('toggle autoscrolling')
-      // cy.spy(reporter, 'stopRunning').as('stop tests clicked')
 
-      // store.start({
-      //   ...defaultStats,
-      //   numberOfPassed: 2,
-      //   startTime: Date.now()
-      // })
+const props = {
+  runState: 'running',
+  stats: {
+    failed: 1,
+    passed: 9,
+    pending: 2,
+  },
+}
 
-      // setTimeout(() => {
-      //   store.stop()
-      // }, timeout)
+it('is renders the timer and can be played/paused', () => {
+  const onRestartSpy = cy.spy().as('onRestart')
+  const onPauseSpy = cy.spy().as('onPause')
+  const wrapper = h({
+    // The parent component will provide runState
+    data() {
+      return {
+        runState: 'running'
+      }
     },
-    
+    setup: () => ({ statsStore: useStatsStore() }),
     render() {
-      return h('div', { style }, h(ReporterHeader))
+      const _this = this
```

Comment:
It took me a while to figure out what's happening here - `this` refers to the Vue instance `this`. Generally, I think mixing the Options and Composition API is a little confusing. What about writing this just using `setup`? This is working for me:

```js
it('is renders the timer and can be played/paused', () => {
  const onRestartSpy = cy.spy().as('onRestart')
  const onPauseSpy = cy.spy().as('onPause')

  const Subject = defineComponent({
    setup() {
      const runState = ref('running')
      const statsStore = useStatsStore() 

      return () => h(ReporterHeader, {
        ...props,
        runState: runState.value,
        onPause() {
          console.log('onPause')
          onPauseSpy()
          runState.value = 'paused'
          statsStore.stop()
        },
        onRestart() {
          onRestartSpy()
          runState.value = 'running'
          statsStore.$reset()
          statsStore.start()
        }
      })
    }
  })

  const wrapper = h(Subject)

  cy.mount(wrapper)
    .get('[data-cy=play-pause-toggle]').as('playPause')
    .click()
    .get('@onPause').should('have.been.called')
    .get('@playPause').click()
    .get('@onRestart').should('have.been.called')
    .get('@playPause').click()
})
```

Reply:
I've had a similar experience when writing TSX/render functions in the composition API - it works, just not always as you'd expect on the first try. I think the types (in Vue core and/or the JSX plugin) probably can be improved to help catch some of the bugs. Eg:

```js
const runState = ref('running')

return () => h(ReporterHeader, {
  runState: runState,
})
```

This should raise an error - the `runState` prop should be a string, but I'm passing a `ref`. Since there is not intelligence around `vue` files, Volar doesn't seem to know about the `ReporterHeader` prop types, and can't tell me to do `.value`. I think this is probably possible via VTI and Volar, just need someone to implement it.